explainability-service/src/main/java/org/kie/trustyai/service/endpoints/explainers/SHAPEndpoint.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.trustyai.service.endpoints.explainers;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.logging.Logger;
+import org.kie.trustyai.explainability.local.shap.ShapConfig;
+import org.kie.trustyai.explainability.local.shap.ShapKernelExplainer;
+import org.kie.trustyai.explainability.model.Prediction;
+import org.kie.trustyai.explainability.model.PredictionInput;
+import org.kie.trustyai.explainability.model.PredictionProvider;
+import org.kie.trustyai.service.config.ServiceConfig;
+import org.kie.trustyai.service.data.DataSource;
+import org.kie.trustyai.service.payloads.BaseExplanationRequest;
+import org.kie.trustyai.service.payloads.BaseExplanationResponse;
+import org.kie.trustyai.service.payloads.SaliencyExplanationResponse;
+
+@Tag(name = "SHAP Explainer Endpoint", description = ".")
+@Path("/explainers/shap")
+public class SHAPEndpoint extends ExplainerEndpoint {
```

Comment:
Minor comment, we will need to verify if the ADR that covers extending TrustyAI can be applied to explainability endpoints too